File: frontend/src/types/agent.ts
export interface AgentConversation {
  id: number
  title: string
  summary?: string | null
  created_at: string
  updated_at: string
  last_message_preview?: string | null
}

export type AgentRole = 'user' | 'assistant' | 'tool'

export interface AgentMessage {
  id: number
  conversation_id: number
  role: AgentRole
  content: string
  tool_name?: string | null
  tool_call_id?: string | null
  tool_args?: Record<string, unknown> | null
  tool_output?: Record<string, unknown> | null
  created_at: string
}

export interface AgentChatResponse {
  conversation_id: number
  messages: AgentMessage[]
}

export interface AgentStreamStatus {
  status: string
  conversation_id?: number | null
  tool_name?: string | null
  tool_call_id?: string | null
  tool_args?: Record<string, unknown> | null
  label?: string | null
  message?: string | null
  tool_output_preview?: string | null
}

File: backend/app/schemas/agent.py
"""Schemas for agent chat and conversation APIs."""

from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ConversationSummary(BaseModel):
    """Lightweight conversation info for lists."""

    id: int
    title: str
    summary: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    last_message_preview: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ConversationListResponse(BaseModel):
    items: List[ConversationSummary]


class ChatMessageRequest(BaseModel):
    """Incoming chat message payload."""

    content: str = Field(..., min_length=1, max_length=4000)
    conversation_id: Optional[int] = None


class ChatMessage(BaseModel):
    id: int
    conversation_id: int
    role: str
    content: str
    tool_name: Optional[str] = None
    tool_call_id: Optional[str] = None
    tool_args: Optional[dict] = None
    tool_output: Optional[dict] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ChatResponse(BaseModel):
    conversation_id: int
    messages: List[ChatMessage]


class MessageListResponse(BaseModel):
    items: List[ChatMessage]
